pkg/security/sasl.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package security
+
+import (
+	"strings"
+
+	"github.com/Shopify/sarama"
+	"github.com/pingcap/errors"
+)
+
+// SASLMechanism defines SASL mechanism.
+type SASLMechanism string
+
+// The mechanisms we currently support.
+const (
+	// UnknownMechanism means the SASL mechanism is unknown.
+	UnknownMechanism SASLMechanism = ""
+	// PlainMechanism means the SASL mechanism is plain.
+	PlainMechanism SASLMechanism = sarama.SASLTypePlaintext
+	// SCRAM256Mechanism means the SASL mechanism is SCRAM-SHA-256.
+	SCRAM256Mechanism SASLMechanism = sarama.SASLTypeSCRAMSHA256
+	// SCRAM512Mechanism means the SASL mechanism is SCRAM-SHA-512.
+	SCRAM512Mechanism SASLMechanism = sarama.SASLTypeSCRAMSHA512
+	// GSSAPIMechanism means the SASL mechanism is GSSAPI.
+	GSSAPIMechanism SASLMechanism = sarama.SASLTypeGSSAPI
+)
+
+// SASLMechanismFormString converts the string to SASL mechanism.
+func SASLMechanismFormString(s string) (SASLMechanism, error) {
```

Comment:
s/Form/From

Reply:
Good catch! 
Thanks for your review!  💚 💙 💜 💛 ❤️